ballerina/tests/12-custom-type-varray-test.bal

```diff
@@ -113,30 +113,247 @@ isolated function insertVarrayWithEmptyArray() returns sql:Error? {
+}
+isolated function insertVarrayWithMixElements() returns sql:Error? {
+    string?[] charArray = [(), "Hello"];
+    byte[]?[] byteArray = [(), [1, 2]];
+    int?[] intArray = [(), 1];
+    boolean?[] boolArray = [(), true];
+    float?[] floatArray = [(), 34];
+    decimal?[] decimalArray = [(), 34.2];
+
+    VarrayValue charVarray = new({ name:"CharArrayType", elements: charArray });
+    VarrayValue byteVarray = new({ name:"ByteArrayType", elements: byteArray });
+    VarrayValue intVarray = new({ name:"IntArrayType", elements: intArray });
+    VarrayValue boolVarray = new({ name:"BoolArrayType", elements: boolArray });
+    VarrayValue floatVarray = new({ name:"FloatArrayType", elements: floatArray });
+    VarrayValue decimalVarray = new({ name:"DecimalArrayType", elements: decimalArray });
+
+    sql:ParameterizedQuery insertQuery = `insert into TestVarrayTable(
+          COL_CHARARR, COL_BYTEARR, COL_INTARR, COL_BOOLARR, COL_FLOATARR, COL_DECIMALARR)
+          values(${charVarray}, ${byteVarray}, ${intVarray}, ${boolVarray}, ${floatVarray}, ${decimalVarray})`;
+    sql:ExecutionResult result = check executeQuery(insertQuery);
+    test:assertExactEquals(result.affectedRowCount, 1, "Affected row count is different.");
+    var insertId = result.lastInsertId;
+    test:assertTrue(insertId is string, "Last Insert id should be string");
+}
+
+// insert direct array types
+@test:Config {
+   groups:["custom-varray"],
+   dependsOn:[insertVarrayWithNullElements]
+}
+isolated function insertDirectArrayTypes() returns sql:Error? {
+    string[] charArray = ["Hello", "World"];
+    byte[] byteArray = [1, 2, 3];
+    byte[][] arrayOfByteArray = [[4, 23, 12]];
+    int[] intArray = [3, 4, 5];
+    boolean[] boolArray = [true, false, false];
+    float[] floatArray = [34, -98.23, 0.981];
+    decimal[] decimalArray = [34, -98.23, 0.981];
+    sql:ParameterizedQuery insertQuery = `insert into TestVarrayTable(COL_CHARARR) values(${charArray})`;
+    sql:ExecutionResult|sql:Error result = executeQuery(insertQuery);
+    if result is sql:ApplicationError {
+       test:assertTrue(result.message().includes("Unsupported varray type :string[] passed as an array type"));
+    } else {
+       test:assertFail("Database Error expected.");
+    }
+    insertQuery = `insert into TestVarrayTable(COL_BYTEARR) values(${arrayOfByteArray})`;
+    result = executeQuery(insertQuery);
+    if result is sql:ApplicationError {
+       test:assertTrue(result.message().includes("Unsupported varray type :byte[][] passed as an array type"));
+    } else {
+       test:assertFail("Database Error expected.");
+    }
+    insertQuery = `insert into TestVarrayTable(COL_BYTEARR) values(${byteArray})`;
+    result = executeQuery(insertQuery);
+    if result is sql:DatabaseError {
+       test:assertTrue(result.message().includes("Error while executing SQL query: insert into " +
+       "TestVarrayTable(COL_BYTEARR) values( ? ). ORA-00600: internal error code"));
```

Comment:
error message is not meaningful

Reply:
This error message in triggered in sql module level. need to change the error message in sql module level.